08-components-lab/src/02-numberguessing.jsx

```diff
@@ -1,17 +1,51 @@
-/**
- * Lab 02:
- *
- * Implement a number guessing game component.
- *
- * The component should provide an input field for the user to guess a
- * number picked randomly by the computer (from 1 to 1,000).
- * When typing a number, the component should print "too low" or "too high" 
- * as a hint to the user.
- *
- * To make the game interesting, computer should sometimes cheat and print wrong messages.
- */
-
-import React from 'react';
-
-export default React.createClass({
-});
+/**
+ * Lab 02:
+ *
+ * Implement a number guessing game component.
+ *
+ * The component should provide an input field for the user to guess a
+ * number picked randomly by the computer (from 1 to 1,000).
+ * When typing a number, the component should print "too low" or "too high"
+ * as a hint to the user.
+ *
+ * To make the game interesting, computer should sometimes cheat and print wrong messages.
+ */
+
+import React from 'react';
+
+export default class NumberGuessing extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            compGuess: Math.floor(Math.random() * (1000)) + 1,
+            userGuess:"0",
+            msg:""
+        }
+    }
+
+    checkGuess = (evt) => {
+        var answer = evt.target.value > this.state.compGuess ? "Too big... try again" : "Too small... try again";

```

Comment:
notice that evt.target.value > this.state.compGuess is repeated twice. Consider extracting this logic inside a const and reusing it in both places.

Also you can merge this two ifs into one. Think of a way to do so.